shared/macros-oval.jinja

```diff
@@ -10,23 +10,36 @@
     - application (String): The application which the configuration file is being checked. Can be any value and does not affect the OVAL check.
     - multi_value (boolean): If set, it means that the parameter can accept multiple values and the expected value must be present in the current list of values.
     - missing_config_file_fail (boolean): If set, the check will fail if the configuration is not existent in the system.
+    - supported_platforms (String List): List of supported platforms the OVAL will applicable to. (e.g.: ['rhel7', 'rhel8', 'fedora'])
 #}}
-{{%- macro oval_check_config_file(path='', prefix_regex='^\s*', parameter='', separator_regex='(?-i)[\s]+', value='', default_pass='', application='', multi_value='', missing_config_file_fail=false) -%}}
+{{%- macro oval_check_config_file(path='', prefix_regex='^\s*', parameter='', separator_regex='(?-i)[\s]+', value='', default_pass='', application='', multi_value='', missing_config_file_fail=false, supported_platforms=['all']) -%}}
 <def-group>
   <definition class="compliance" id="{{{ rule_id }}}" version="1">
     <metadata>
       <title>{{{ rule_title }}}</title>
       <affected family="unix">
-        <platform>multi_platform_all</platform>
+{{#-
+  Use macro oval_affected(products) when it is available
+#}}
+        <platform>multi_platform_rhel</platform>
+        <platform>multi_platform_fedora</platform>
+        <platform>multi_platform_rhv</platform>
+        <platform>multi_platform_ol</platform>
       </affected>
       <description>{{{ rule_description }}}</description>
     </metadata>
     {{%- if missing_config_file_fail %}}
     <criteria comment="{{{ application }}} is configured correctly and configuration file exists"
     operator="AND">
     {{%- endif %}}
+    {{%- if application == "sshd" %}}
+    <criteria comment="{{{ application }}} is configured correctly or is not installed"
+    operator="OR">
+        {{{- application_not_required_or_requirement_unset() }}}
+    {{%- else %}}
```

Comment:
If I would be using macro `oval_check_config_file`, how would I make it check if a package is installed?
Should there be an `extend_definition` in the else clause to check if the package is installed?

Reply:
The idea was to avoid the OVAL check to test if the package is installed or not and let this be handled by rule applicability (e.g.: CPE dictionaries). Is it really needed?

Reply:
I see, that makes sense.